pkg/controller/utils/pod/pod.go

```diff
@@ -158,17 +163,44 @@ func IsCannotStartReason(reason string) bool {
 
 // CannotStart returns true if the Pod is currently experiencing abnormal start condition
 func CannotStart(pod *v1.Pod) (bool, datadoghqv1alpha1.ExtendedDaemonSetStatusReason) {
-	for _, s := range pod.Status.ContainerStatuses {
+	for _, s := range containerStatusList(pod) {
 		if s.State.Waiting != nil && IsCannotStartReason(s.State.Waiting.Reason) {
-			return true, datadoghqv1alpha1.ExtendedDaemonSetStatusReason(s.State.Waiting.Reason)
+			return true, convertReasonToEDSStatusReason(s.State.Waiting.Reason)
 		}
 	}
 	return false, datadoghqv1alpha1.ExtendedDaemonSetStatusReasonUnknown
 }
 
+func convertReasonToEDSStatusReason(reason string) datadoghqv1alpha1.ExtendedDaemonSetStatusReason {
```

Comment:
This is 1:1 by most part, replace with a string to bool map lookup maybe?

Reply:
@xornivore what do you think about my change?